### PR TITLE
no summaries when there is no request for tensorboard

### PR DESCRIPTION
**Pull request by tomjur**

### CODE REVIEW

File: v1_embedding/model_trainer.py [hunk 1]
```diff
@@ -221,10 +221,11 @@ def do_train_loop(self):
                         summary_writer_train.add_summary(train_summaries, global_step=global_step)
                     if (global_step % self.operational_config['validation_batch_frequency']) == 1:
                         for validation_batch in self.batch_iterator_validation:
-                            validation_summaries = self.do_validation_batch(sess, global_step, epoch_num, batch_index,
-                                                                            validation_batch, print_to_file=False)
-                            if validation_summaries and use_tensorboard:
-                                summary_writer_validation.add_summary(validation_summaries, global_step=global_step)
+                            if use_tensorboard:
```
Comment (ashual) [inline v1_embedding/model_trainer.py]: don't you want it to be before the for loop?

File: v1_embedding/gan_model.py [hunk 2]
```diff
@@ -119,10 +122,12 @@ def __init__(self, config_file, operational_config_file, embedding_handler):
                                       false_fn=lambda: tf.constant(False)
                                   )),
                 self._total_steps_counter.update
-            )
+            ) if self.do_tensorboard else None
 
             # after appropriate train step is executed, we notify the policy and count for tensorboard
-            with tf.control_dependencies([policy_selected_train_step, counter_steps]):
+            control_list = [policy_selected_train_step, counter_steps] if self.do_tensorboard else [
```
Comment (ashual) [inline v1_embedding/gan_model.py]: just for common knowledge, what is this control_dependencies thing?

File: v1_embedding/model_trainer.py (hunk 1 above)
Comment (tomjur) [reply v1_embedding/model_trainer.py]: sure - good idea

File: v1_embedding/gan_model.py (hunk 2 above)
Comment (tomjur) [reply v1_embedding/gan_model.py]: forces the steps in the list to be executed before the steps inside the with clause.

**Pull request merged**